lib/dataSource.js

```diff
@@ -92,19 +94,40 @@ kettle.dataSource.priorities = {
     CouchDB: 90
 };
 
-kettle.dataSource.parse = function (string) {
+kettle.dataSource.JSONParseErrors = [];
+
+kettle.dataSource.accumulateJSONError = function (str, hash) {
+    var error = "JSON parse error at line "+ hash.loc.first_line + ", col " + hash.loc.last_column + ", found: \'"+ hash.token + "\' - expected: "+ hash.expected.join(", ");
+    kettle.dataSource.JSONParseErrors.push(error);
+};
+
+// Adapt to shitty integration model of JISON-based parsers - beware that any other user of this module will find it permanently corrupted
+// TODO: Unfortunately the parser has no error recovery states - this can only ever accumulate a single error
+jsonlint.parser.parseError = jsonlint.parser.lexer.parseError = kettle.dataSource.accumulateJSONError;
```

Comment:
If you are seeing shortcomings with JSONlint, it could be worth considering [AJV](https://github.com/epoberezkin/ajv).  It can be instructed to continue past the first error, and is also evolving closely in parallel with the standard itself.